### PR TITLE
Fix clicks not occurring if drag events weren't handled

### PR DESCRIPTION
**Pull request by ekrctb**
- Resolves ppy/osu#2862.

TestCaseMouseStates is modified to add tests for this behavior.

---

The recent rework of InputManager overlooked drag-clicking logic for the case where the object being "dragged" didn't handle the drag. The objects expect to receive click events in this case, as per the previous functionality.

### CODE REVIEW

File: osu.Framework/Input/InputManager.cs [hunk 1]
```diff
@@ -426,12 +426,7 @@ public virtual void HandleMouseButtonStateChange(InputState state, MouseButton b
 
                 if (button == MouseButton.Left)
                 {
-                    if (isDragging)
-                    {
-                        isDragging = false;
-                        handleMouseDragEnd(state);
-                    }
-                    else
+                    if (DraggedDrawable == null || Vector2Extensions.Distance(mouse.PositionMouseDown ?? mouse.Position, mouse.Position) <= click_drag_distance)
```
Comment (peppy) [inline osu.Framework/Input/InputManager.cs]: This conditional seems to be wrong. The position check creates incorrect behaviour when dragging a clickable item inside a `ScrollContainer`, for instance.

Comment (smoogipoo) [reply osu.Framework/Input/InputManager.cs]: Expand on this a bit more? Where are you seeing this + what you'd expect? I've tested in song select and leaderboard and it seems fine to me?

Comment (peppy) [reply osu.Framework/Input/InputManager.cs]: screencast 2018-06-25 12-27-39

Comment (peppy) [reply osu.Framework/Input/InputManager.cs]: On further testing, it seems to work fine just removing the distance check part of the conditional. Tests will need to be updated in line with this.

I'm going to a commit a proposed fix (along with a few clean-ups to drag handling conditionals) on this branch.

**Pull request merged**